### DISCUSSION TITLE
Recommended approach for returning JWT token with /auth/register response

### DISCUSSION THREAD

**Discussion by cmcilvoy** [fastapi-users/fastapi-users#575] [Q&A]
What is the best approach for returning a JWT token after POST to /auth/register response to prevent user from having to authenticate again?

Basically add a JWT token to this:
{ id: "e88f156c-9aef-4077-bd5e-1a38627cd1c9", email: "test-4@nodomain.com", is_active: true, is_superuser: false, is_verified: false, first_name: null, last_name: null, display_name: null }

**Comment by frankie567** [comment]
What I usually do in the frontend app is to call `/login` right after `/register`; you still have the password at hand at that moment so it should be quite straightforward.

Other than that, if you wish to have the token in the register response, you can implement the route yourself:
* You can take inspiration from the built-in one : https://github.com/frankie567/fastapi-users/blob/master/fastapi_users/router/register.py
* To get the JWT token, you can use the `_generate_token` method of the `JWTAuthentication` class.
